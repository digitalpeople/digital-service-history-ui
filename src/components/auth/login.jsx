import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import TextInput from '../forms/text-input';
import { loginAction } from '../../store/auth/auth.action';
import './login.scss';

const Login = (props) => {
  const {
    onLogin,
    auth,
  } = props;
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  if (auth.isAuthenticated) {
    return <Redirect to="/about" />;
  }

  return (
    <section className="login">
      <header>Login</header>
      <article className="container">
        <TextInput
          name="username"
          label="Username"
          onChange={value => setUsername(value)}
        />

        <TextInput
          name="password"
          label="Password"
          onChange={value => setPassword(value)}
          isPassword
        />

        <button
          type="submit"
          className="btn btn-primary btn-block"
          onClick={() => onLogin({
            username,
            password,
          })}
        >
          Login
        </button>

      </article>
    </section>
  );
};

Login.propTypes = {
  auth: PropTypes.shape({
    isAuthenticated: PropTypes.bool.isRequired,
  }).isRequired,
  onLogin: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth,
});

const mapDispatchToProps = dispatch => ({
  onLogin: ({ username, password }) => dispatch(loginAction({ username, password })),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
