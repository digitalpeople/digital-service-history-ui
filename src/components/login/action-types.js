export const login = '@auth/LOGIN';
export const logout = '@auth/LOGOUT';
