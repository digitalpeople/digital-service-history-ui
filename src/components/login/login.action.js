import {
  login,
  logout,
} from './action-types';

export const loginAction = ({ username, password }) => ({
  type: login,
  username,
  password,
});

export const logoutAction = () => ({
  type: logout,
});
