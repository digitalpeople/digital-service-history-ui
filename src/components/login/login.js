import React from 'react';

const Login = () => (
  <section className="login">
    <h2>Login</h2>
    <article className="container">
      <label htmlFor="username">
        Username
        <input
          type="text"
          placeholder="Username"
          name="username"
          required
        />
      </label>

      <label htmlFor="password">
        Password
        <input
          type="password"
          placeholder="Password"
          name="password"
          required
        />
      </label>

      <button type="submit">Login</button>

    </article>
  </section>
);

export default Login;
