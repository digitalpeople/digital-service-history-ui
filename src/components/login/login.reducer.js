import {
  login,
  logout,
} from './action-types';
import initialState from './login.state';

const loginReducer = (state = initialState, action) => {
  switch (action.type) {
  case login: {
    return {
      ...state,
      login: true,
    };
  }
  case logout:
    return initialState;

  default:
    return state;
  }
};

export default loginReducer;
