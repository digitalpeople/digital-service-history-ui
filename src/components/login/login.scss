@import "../../styles/variables.scss";

.login {

  button {
    padding: 10px 14px;
    background-color: #03a9f4;
    font-size: 14px;
    outline: 0;
    outline-offset: 0;
    border: 0;
    border-radius: 2px;
    font-weight: 400;
    line-height: 1.1;
    text-transform: uppercase;
  }
}
