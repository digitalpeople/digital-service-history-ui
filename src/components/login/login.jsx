import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import TextInput from '../forms/text-input';
import { loginAction } from './login.action';
import './login.scss';

const Login = (props) => {
  const {
    onLogin,
  } = props;
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  return (
    <section className="login">
      <header>Login</header>
      <article className="container">
        <TextInput
          name="username"
          label="Username"
          onChange={value => setUsername(value)}
        />

        <TextInput
          name="password"
          label="Password"
          onChange={value => setPassword(value)}
          isPassword
        />

        <button
          type="submit"
          className="btn btn-primary btn-block"
          onClick={() => onLogin({
            username,
            password,
          })}
        >
          Login
        </button>

      </article>
    </section>
  );
};

Login.propTypes = {
  onLogin: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  login: state.login,
});

const mapDispatchToProps = dispatch => ({
  onLogin: ({ username, password }) => dispatch(loginAction({ username, password })),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
