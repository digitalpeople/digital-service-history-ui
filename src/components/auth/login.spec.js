import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import { Login } from './login';

configure({ adapter: new Adapter() });

describe('Login', () => {
  it('should render component', () => {
    // ASSIGN
    const props = {
      onLogin: jest.fn(),
      auth: {
        isAuthenticated: false,
      },
    };

    // ACT
    const actual = shallow(<Login {...props} />);

    // ASSERT
    expect(toJSON(actual)).toMatchSnapshot();
  });
  it('should redirect component', () => {
    // ASSIGN
    const props = {
      onLogin: jest.fn(),
      auth: {
        isAuthenticated: true,
      },
    };

    // ACT
    const actual = shallow(<Login {...props} />);

    // ASSERT
    expect(toJSON(actual)).toMatchSnapshot();
  });
});
