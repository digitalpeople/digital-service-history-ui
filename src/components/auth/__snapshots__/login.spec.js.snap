// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render component 1`] = `
<section
  className="login"
>
  <header>
    Login
  </header>
  <article
    className="container"
  >
    <TextInput
      isPassword={false}
      label="Username"
      name="username"
      onChange={[Function]}
    />
    <TextInput
      isPassword={true}
      label="Password"
      name="password"
      onChange={[Function]}
    />
    <button
      className="btn btn-primary btn-block"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </article>
</section>
`;
